### PR TITLE
Adjust clang-tidy config and fix warnings

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

File: examples/example_vm/example_vm.cpp [hunk 1]
```diff
@@ -52,15 +52,15 @@ evmc_capabilities_flagset get_capabilities(evmc_vm* /*instance*/)
...
+    auto* vm = static_cast<ExampleVM*>(instance);
     if (std::strcmp(name, "verbose") == 0)
     {
         if (value == nullptr)
             return EVMC_SET_OPTION_INVALID_VALUE;
 
         char* end = nullptr;
-        long int v = std::strtol(value, &end, 0);
-        if (end == value)  // Parsing the value failed.
+        long int v = std::strtol(value, &end, 0);  // NOLINT(google-runtime-int)
```
Comment (gumb0) [inline examples/example_vm/example_vm.cpp:62]: Maybe `auto`?

File: examples/example_vm/example_vm.cpp [hunk 2]
```diff
@@ -74,7 +74,7 @@ enum evmc_set_option_result set_option(evmc_vm* instance, const char* name, cons
 /// The Example VM stack representation.
 struct Stack
 {
-    evmc_uint256be items[1024];       ///< The array of stack items, uninitialized.
+    evmc_uint256be items[1024]{};     ///< The array of stack items, uninitialized.
```
Comment (gumb0) [inline examples/example_vm/example_vm.cpp:77]: This makes them initialized, so comment outdated?

File: examples/example_vm/example_vm.cpp [hunk 3]
```diff
@@ -159,15 +159,15 @@ inline evmc_address to_address(evmc_uint256be value)
 
 
 /// The example implementation of the evmc_vm::execute() method.
-evmc_result execute(evmc_vm* instance,
+evmc_result execute(evmc_vm* instance,  // NOLINT(readability-function-cognitive-complexity)
```
Comment (gumb0) [inline examples/example_vm/example_vm.cpp:162]: Isn't this check disabled?

File: tools/vmtester/vmtester.cpp [hunk 4]
```diff
@@ -7,7 +7,7 @@
 #include <evmc/loader.h>
 #include <iostream>
 
-evmc::VM evmc_vm_test::owned_vm;
+evmc::VM evmc_vm_test::owned_vm;  // NOLINT(cppcoreguidelines-avoid-non-const-global-variables)
```
Comment (gumb0) [inline tools/vmtester/vmtester.cpp:10]: Isn't this check disabled?

File: tools/evmc/main.cpp [hunk 5]
```diff
@@ -39,89 +39,95 @@ struct HexValidator : public CLI::Validator
...
+        run_cmd.add_option("--rev", rev, "EVM revision", true);
+        run_cmd.add_option("--input", input_arg, "Input bytes")->check(Hex | CLI::ExistingFile);
+        run_cmd.add_flag(
+            "--create", create,
+            "Create new contract out of the code and then execute this contract with the input");
+        run_cmd.add_flag(
+            "--bench", bench,
+            "Benchmark execution time (state modification may result in unexpected behaviour)");
+
+        try
```
Comment (gumb0) [inline tools/evmc/main.cpp:78]: Why do you need to nest it, seems it could be a single outer try-block for everything?

Comment (chfast) [reply tools/evmc/main.cpp:78]: In the inner `try-catch`, in the `catch (const CLI::ParseError& e)` the `app` object is used.

**Review by gumb0 — APPROVED** [review]

**Pull request merged**